sqlassignment3.txt

```diff
@@ -0,0 +1,353 @@
+	Extract the middle 3 characters from the string 'ABCDEFG'.
+
+declare @string varchar(7)='ABCDEFG'
+declare @lengt int =((len(@string)-2)/2)+1
+select substring(@string,@lengt,3);
+
+ 	From a table 'Employees' with a column 'FullName', write a query to extract the first name 
+
+select left(fullname,CHARINDEX(' ',fullname)) as firstname from employees
+
+
+	Extract the first 5 characters from the string 'SQL Server 2022'.
+
+declare @string varchar(15)='SQL Server 2022'
+select substring(@string,1,5);
+
+	From a 'Products' table with a 'ProductCode' column, write a query to get the first 3 characters of each product code.
+
+select SUBSTRING(ProductCode,1,3) as Productcode@3 from Products
+
+	Extract the last 4 characters from the string 'ABCDEFGHIJKL'.
+
+declare @string varchar(15)='ABCDEFGHIJKL'
+select Right(@string,4);
+
+ 	From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+
+select right(orderid,4) as NNNN from orders
```

Comment:
Try to write this query generically, for example order is ORD-YYYY-1234, ORD-YYYY-12, ORD-YYYY-123432344

Reply:
select replace(orderid,'ORD-YYYY-','') as NNNN from orders

modified using replace()

Reply:
I think my example was wrong, consider the order ID's as ORD-123-1234, ORD-122-12, ORD-0922334-123432344. so how can we get the last numbers.

Reply:
select right(orderid,CHARINDEX('-',reverse(orderid))-1) from orders;